src/Symfony/Component/Translation/Extractor/Visitor/ConstraintVisitor.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Extractor\Visitor;
+
+use PhpParser\Node;
+use PhpParser\NodeVisitor;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ *
+ * Code mostly comes from https://github.com/php-translation/extractor/blob/master/src/Visitor/Php/Symfony/Constraint.php
+ */
+final class ConstraintVisitor extends AbstractVisitor implements NodeVisitor
+{
+    public const CONSTRAINT_CLASS_NAMES = [
+        'AbstractComparison',
+        'All',
+        'Bic',
+        'Blank',
+        'Callback',
+        'CardScheme',
+        'Choice',
+        'Collection',
+        'Composite',
+        'Count',
+        'Country',
+        'Currency',
+        'Date',
+        'DateTime',
+        'DisableAutoMapping',
+        'DivisibleBy',
+        'Email',
+        'EnableAutoMapping',
+        'EqualTo',
+        'Existence',
+        'Expression',
+        'File',
+        'GreaterThan',
+        'GreaterThanOrEqual',
+        'GroupSequence',
+        'GroupSequenceProvider',
+        'Iban',
+        'IdenticalTo',
+        'Image',
+        'Ip',
+        'Isbn',
+        'IsFalse',
+        'IsNull',
+        'Issn',
+        'IsTrue',
+        'Json',
+        'Language',
+        'Length',
+        'LessThan',
+        'LessThanOrEqual',
+        'Locale',
+        'Luhn',
+        'Negative',
+        'NegativeOrZero',
+        'NotBlank',
+        'NotCompromisedPassword',
+        'NotEqualTo',
+        'NotIdenticalTo',
+        'NotNull',
+        'NumberConstraintTrait',
+        'Optional',
+        'Positive',
+        'PositiveOrZero',
+        'Range',
+        'Regex',
+        'Required',
+        'Time',
+        'Timezone',
+        'Traverse',
+        'Type',
+        'Unique',
+        'Url',
+        'Uuid',
+        'Valid',
+    ];
```

Comment:
Is there any more sustainable solution to have (automatically?) the whole list of constraints here?

Reply:
Pass the list as constructor argument and compute it in a compiler pass?

Reply:
Done. But I'm not 100% sure about how I did it